spring-ai-core/src/main/kotlin/org/springframework/ai/chat/client/ChatClientExtensions.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2023-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.chat.client
+
+/**
+ * Extension for [ChatClient] providing a `entity<Foo>()` variant.
+ *
+ * @author Josh Long
+ */
+
+inline fun <reified T> ChatClient.CallResponseSpec.entity(): T = entity(T::class.java) as T
```

Comment:
@joshlong Could you please use `ParameterizedTypeReference` instead of `Class` in order to avoid type erasure?